File: tests/fixtures/py_inheritance/child.py
from base import Base


# Child.run() appelle self.shared() : 'shared' n'existe pas dans Child, la resolution
# doit REMONTER l'heritage jusqu'a Base et emettre child.py::run -> base.py::shared
# (methode heritee vivant dans un autre fichier).
class Child(Base):
    def run(self):
        return self.shared()

File: tests/fixtures/ts_extends/child.ts
import { Base } from './base';

// Child.run() appelle this.shared() : la resolution doit suivre la clause `extends Base`
// et emettre child.ts::run -> base.ts::shared (methode heritee dans un autre fichier).
export class Child extends Base {
  run() {
    return this.shared();
  }
}
